src/hotspot/share/compiler/compilerOracle.cpp

```diff
@@ -308,26 +304,15 @@ static void register_command(TypedMethodOptionMatcher* matcher,
 }
 
 template<typename T>
-bool CompilerOracle::has_option_value(const methodHandle& method, enum CompileCommand option, T& value, bool verify_type) {
+bool CompilerOracle::has_option_value(const methodHandle& method, enum CompileCommand option, T& value) {
   enum OptionType type = option2type(option);
   if (type == OptionType::Unknown) {
```

Comment:
Could this be an assertion done inside option_matches_type?

Reply:
I move it out to option_matches_type so that it won't affect the hot path. I did have to add an extra check to the compiler_directives_init that uses has_option_value and sent OptionType::Unknown for options that don't have a command. But better have it there than in the common path.

Reply:
Good!